lib/Doctrine/Annotations/TypeParser/PHPStanTypeParser.php

```diff
@@ -165,6 +165,8 @@ private function resolveIdentifierNode(TypeNode $typeNode, Scope $scope) : Type
                 return new FloatType();
             case 'string':
                 return new StringType();
+            case 'array':
+                return new ListType(new MixedType());
```

Comment:
This should be an arbitrary map instead.

Reply:
with strings as keys? Map expects scalars as keys, so I cannot put mixed there. I'm not sure if this won't break validation.

Reply:
you can put union of string and int there

Reply:
Union is still not a Scalar, do we want to allow any `Type` as keyType and check type using assertion?

Reply:
Is this what you meant? https://github.com/Majkl578/doctrine-annotations/pull/2/commits/b274cbda904ef7589f9d238f3a22bf0cd1cb1516

Reply:
👍